plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -358,3 +333,26 @@ export function addSqlCommenterComment(span: Span, query: string): string {
 
   return `${query} /*${commentString}*/`;
 }
+
+/**
+ * Attempt to get a message string from a thrown value, while being quite
+ * defensive, to recognize the fact that, in JS, any kind of value (even
+ * primitives) can be thrown.
+ */
+export function getErrorMessage(e: unknown) {
+  return typeof e === 'object' && e !== null && 'message' in e
+    ? String((e as { message?: unknown }).message)
+    : undefined;
+}
+
+export function isObjectWithTextString(it: unknown): it is ObjectWithText {
+  return (
+    typeof it === 'object' &&
+    typeof (it as null | { text?: unknown })?.text === 'string'
+  );
+}
+
+export type ObjectWithText = {
+  text: string;
+  [k: string]: unknown;
+};
```

Comment:
nit: this might be more appropriate in `internal-types.ts`

Reply:
This type is pretty tightly-coupled to exactly what's being verified in the `isObjectWithTextString`, and we need to make sure the type and the predicate function stay in sync, so I think it makes sense to keep them together. Other code also probably shouldn't depend on this type directly; it's just here to help TS narrow. Once TS is updated, this type won't even need to be named or exported, as we'll be able to remove all the `as ObjectWithText` casts, because TS will correctly infer that automatically. I.e., it'll be possible to just have:

```ts
export function isObjectWithTextString(it: unknown): it is { text: string, [k: string]: unknown } {
  // ...
}
```